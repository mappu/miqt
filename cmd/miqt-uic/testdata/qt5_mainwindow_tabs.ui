<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="1" column="0">
     <widget class="QTabWidget" name="tabWidget_3">
      <property name="tabPosition">
       <enum>QTabWidget::West</enum>
      </property>
      <widget class="QWidget" name="tab_5">
       <attribute name="title">
        <string>Tab 1</string>
       </attribute>
      </widget>
      <widget class="QWidget" name="tab_6">
       <attribute name="title">
        <string>Tab 2</string>
       </attribute>
      </widget>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>First Tab</string>
       </attribute>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Second Tab</string>
       </attribute>
      </widget>
     </widget>
    </item>
    <item row="2" column="1">
     <widget class="QTabWidget" name="tabWidget_2">
      <property name="tabPosition">
       <enum>QTabWidget::South</enum>
      </property>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>Tab 1</string>
       </attribute>
      </widget>
      <widget class="QWidget" name="tab_4">
       <attribute name="title">
        <string>Tab 2</string>
       </attribute>
      </widget>
     </widget>
    </item>
    <item row="1" column="2">
     <widget class="QTabWidget" name="tabWidget_4">
      <property name="tabPosition">
       <enum>QTabWidget::East</enum>
      </property>
      <widget class="QWidget" name="tab_7">
       <attribute name="title">
        <string>Tab 1</string>
       </attribute>
      </widget>
      <widget class="QWidget" name="tab_8">
       <attribute name="title">
        <string>Tab 2</string>
       </attribute>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>29</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
