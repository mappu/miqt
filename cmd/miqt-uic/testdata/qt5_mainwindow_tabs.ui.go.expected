// Generated by miqt-uic. To update this file, edit the .ui file in
// Qt Designer, and then run 'go generate'.
//
//go:generate miqt-uic go_generate_args

package test_package_name

import (
	"github.com/mappu/miqt/qt"
)

type MainWindowUi struct {
	MainWindow    *qt.QMainWindow
	centralwidget *qt.QWidget
	gridLayout    *qt.QGridLayout
	tabWidget_3   *qt.QTabWidget
	tab_5         *qt.QWidget
	tab_6         *qt.QWidget
	tabWidget     *qt.QTabWidget
	tab           *qt.QWidget
	tab_2         *qt.QWidget
	tabWidget_2   *qt.QTabWidget
	tab_3         *qt.QWidget
	tab_4         *qt.QWidget
	tabWidget_4   *qt.QTabWidget
	tab_7         *qt.QWidget
	tab_8         *qt.QWidget
	menubar       *qt.QMenuBar
	statusbar     *qt.QStatusBar
}

// NewMainWindowUi creates all Qt widget classes for MainWindow.
func NewMainWindowUi() *MainWindowUi {
	ui := &MainWindowUi{}
	ui.MainWindow = qt.NewQMainWindow(nil)
	ui.MainWindow.SetObjectName("MainWindow")
	ui.MainWindow.Resize(800, 600)
	ui.centralwidget = qt.NewQWidget(ui.MainWindow.QWidget)
	ui.centralwidget.SetObjectName("centralwidget")
	ui.gridLayout = qt.NewQGridLayout(ui.centralwidget)
	ui.gridLayout.SetObjectName("gridLayout")
	ui.gridLayout.SetContentsMargins(11, 11, 11, 11)
	ui.gridLayout.SetSpacing(6)
	ui.tabWidget_3 = qt.NewQTabWidget(ui.centralwidget)
	ui.tabWidget_3.SetObjectName("tabWidget_3")
	ui.tabWidget_3.SetTabPosition(qt.QTabWidget__West)
	ui.tab_5 = qt.NewQWidget(ui.tabWidget_3.QWidget)
	ui.tab_5.SetObjectName("tab_5")
	ui.tabWidget_3.AddTab(ui.tab_5, "")
	ui.tab_6 = qt.NewQWidget(ui.tabWidget_3.QWidget)
	ui.tab_6.SetObjectName("tab_6")
	ui.tabWidget_3.AddTab(ui.tab_6, "")

	ui.gridLayout.AddWidget2(ui.tabWidget_3.QWidget, 1, 0)
	ui.tabWidget = qt.NewQTabWidget(ui.centralwidget)
	ui.tabWidget.SetObjectName("tabWidget")
	ui.tab = qt.NewQWidget(ui.tabWidget.QWidget)
	ui.tab.SetObjectName("tab")
	ui.tabWidget.AddTab(ui.tab, "")
	ui.tab_2 = qt.NewQWidget(ui.tabWidget.QWidget)
	ui.tab_2.SetObjectName("tab_2")
	ui.tabWidget.AddTab(ui.tab_2, "")

	ui.gridLayout.AddWidget2(ui.tabWidget.QWidget, 0, 1)
	ui.tabWidget_2 = qt.NewQTabWidget(ui.centralwidget)
	ui.tabWidget_2.SetObjectName("tabWidget_2")
	ui.tabWidget_2.SetTabPosition(qt.QTabWidget__South)
	ui.tab_3 = qt.NewQWidget(ui.tabWidget_2.QWidget)
	ui.tab_3.SetObjectName("tab_3")
	ui.tabWidget_2.AddTab(ui.tab_3, "")
	ui.tab_4 = qt.NewQWidget(ui.tabWidget_2.QWidget)
	ui.tab_4.SetObjectName("tab_4")
	ui.tabWidget_2.AddTab(ui.tab_4, "")

	ui.gridLayout.AddWidget2(ui.tabWidget_2.QWidget, 2, 1)
	ui.tabWidget_4 = qt.NewQTabWidget(ui.centralwidget)
	ui.tabWidget_4.SetObjectName("tabWidget_4")
	ui.tabWidget_4.SetTabPosition(qt.QTabWidget__East)
	ui.tab_7 = qt.NewQWidget(ui.tabWidget_4.QWidget)
	ui.tab_7.SetObjectName("tab_7")
	ui.tabWidget_4.AddTab(ui.tab_7, "")
	ui.tab_8 = qt.NewQWidget(ui.tabWidget_4.QWidget)
	ui.tab_8.SetObjectName("tab_8")
	ui.tabWidget_4.AddTab(ui.tab_8, "")

	ui.gridLayout.AddWidget2(ui.tabWidget_4.QWidget, 1, 2)
	ui.MainWindow.SetCentralWidget(ui.centralwidget) // Set central widget
	ui.menubar = qt.NewQMenuBar(ui.MainWindow.QWidget)
	ui.menubar.SetObjectName("menubar")
	ui.menubar.Resize(800, 29)
	ui.MainWindow.SetMenuBar(ui.menubar)
	ui.statusbar = qt.NewQStatusBar(ui.MainWindow.QWidget)
	ui.statusbar.SetObjectName("statusbar")
	ui.MainWindow.SetStatusBar(ui.statusbar)

	ui.Retranslate()

	ui.tabWidget.SetCurrentIndex(0)

	return ui
}

// Retranslate reapplies all text translations.
func (ui *MainWindowUi) Retranslate() {
	ui.MainWindow.SetWindowTitle(qt.QCoreApplication_Tr("MainWindow"))
	ui.tabWidget_3.SetTabText(ui.tabWidget_3.IndexOf(ui.tab_5), qt.QTabWidget_Tr("Tab 1"))
	ui.tabWidget_3.SetTabText(ui.tabWidget_3.IndexOf(ui.tab_6), qt.QTabWidget_Tr("Tab 2"))
	ui.tabWidget.SetTabText(ui.tabWidget.IndexOf(ui.tab), qt.QTabWidget_Tr("First Tab"))
	ui.tabWidget.SetTabText(ui.tabWidget.IndexOf(ui.tab_2), qt.QTabWidget_Tr("Second Tab"))
	ui.tabWidget_2.SetTabText(ui.tabWidget_2.IndexOf(ui.tab_3), qt.QTabWidget_Tr("Tab 1"))
	ui.tabWidget_2.SetTabText(ui.tabWidget_2.IndexOf(ui.tab_4), qt.QTabWidget_Tr("Tab 2"))
	ui.tabWidget_4.SetTabText(ui.tabWidget_4.IndexOf(ui.tab_7), qt.QTabWidget_Tr("Tab 1"))
	ui.tabWidget_4.SetTabText(ui.tabWidget_4.IndexOf(ui.tab_8), qt.QTabWidget_Tr("Tab 2"))
}
